Doc/library/asyncio-eventloop.rst

```diff
@@ -1046,6 +1046,14 @@ Unix signals
 
    Like :func:`signal.signal`, this function must be invoked in the main
    thread.
+   
+   While signal handlers registered using :func:`signal.signal` will
+   typically be called immediately, handlers registered using this
```

Comment:
That's not true. There are no such guarantees for `signal.signal` callback. E.g. a long-running C function can block Python signals until it completes its execution.

Reply:
I know, but that's why I used the "typically" qualifier and didn't say that immediate callbacks are guaranteed.

I do feel like it's probably safe to say that the typical user experience with :func:`signal.signal` is that there's no perceptible delay, since the same delays would affect the default handlers for Ctrl-C / SIGINT and be very noticeable. I don't know if my experience with delays in callbacks registered with :func:`add_signal_handler` is typical though.

Reply:
> I know, but that's why I used the "typically" qualifier and didn't say that immediate callbacks are guaranteed.

"immediate" is a very strong word that "typically" doesn't really soften :) Again, signal callbacks are not called "immediately" by any typical or atypical definition. That's why I don't like this wording. My 2c.

> no perceptible delay

There might be no perceptible delay, but the delay can still be fractions of a second which is infinitely long from the CPU perspective.

The real reason of this perceptible delay is that signal callbacks in asyncio are executed as all other callbacks - when they go through the queue. And if the callbacks queue is long it might take them longer to execute than ones registered via `signal.signal`. I suggest explaining from this perspective.

Reply:
Interestingly, I noticed some stuff currently in CI that might address it for my use case, which is that I want the program to stop right away when I hit Ctrl-C. The change adds special handling for SIGINT in the loop.

How about wording:

While signal handlers registered using :func:`signal.signal` will typically be called without significant delays as seen from the perspective of an interactive user, signal callbacks in `asyncio` are subject to additional delays while waiting for other work in the given event loop to be completed.

Reply:
I like the new wording!

Reply:
Just noticed that I did the PR from master. Sorry about that.